src/error.rs

```diff
@@ -73,6 +73,7 @@ pub enum WriteError {
     /// If for some reason the implementation of that crate changes, we could
     /// be affected, although this is very unlikely.
     InternalLibWriteError,
+    InvalidUtf8Error(FromUtf8Error),
```

Comment:
This should never happen, and I think a panic is better than an error. We are explicitly serializing UTF-8, and a failure to do that is not recoverable.